phac-nml/irida-next
# GraphQL: CreateSample mutation

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in a CreateSample mutation type to the GraphQL api which allows a User to create samples within projects that they have access to.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com`
1. Open the graphiql interface `http://localhost:3000/graphiql`
1. Query for a Project that you can access.
   * ```graphql
     {
       project(fullPath: "bacillus/bacillus-anthracis/outbreak-2022") {
         id
         name
         path
       }
     }
     ```
1. See that you get a valid result
1. Execute a mutation to create a Sample.
   * ```graphql
     {
       createSample(input: { projectId: ID_VALUE_FROM_ABOVE, name: "Sample Name", description: "Sample Description" }) {
         errors
         sample {
           id,
           name,
           description
         }
       }
     }
     ```
1. See that you get a valid result
1. Repeat mutation above
1. See that you get an error noting that Sample name has already been taken

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
